packages/client-beta/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,21 @@
+import clsx from 'clsx';
+import { Component, JSX, mergeProps, splitProps } from 'solid-js';
+import styles from './Button.module.css';
+
+interface ButtonProps extends JSX.ButtonHTMLAttributes<HTMLButtonElement> {
+  children: string;
+  primary?: boolean;
+}
+
+const Button: Component<ButtonProps> = (_props) => {
+  const props = mergeProps({ primary: false }, _props);
+  const [, buttonAttributes] = splitProps(props, ['children', 'primary']);
```

Comment:
Nie do końca rozumiem dlaczego potrzebujmy używać tutaj merge/splitProps. Nie wystarczy jak zrobimy destrukturyzację od razu i default przypiszemy false do primary zamiast używać mergeProps?
I ja bym to jeszcze zamienił na function, jak pisałem w komentarzu do Input.tsx 
```suggestion
function Button({ children, primary = false, ...buttonAttributes }): Component<ButtonProps> = {
    return (
    <button {...buttonAttributes} class={clsx(styles.button, primary && styles.primary)}>
      {children}
    </button>
  );
}
```

Reply:
Najłatwiej byłoby to przegadać, ale spróbuję też opisać o co chodzi :D. Takie zachowanie zaleca sam solid. Argumentują to tym, że przy destrukturyzacji te rzeczy tracą reaktywność. Dlatego jeśli chcę ustawić defaultową wartość dla jakiegoś propsa to Solid z tego co widziałem proponuje użycie mergeProps. 

SplitProps jest po to, żeby jako w argumenty do buttona nie pakować rzeczy, które nie powinny nimi być. Czyli na przykład jak sobie ustawimy inne customowe propsy w buttonie typu `margin` albo `asLink` (z głowy rzucam przykładami na szybko :D) to nie powinny raczej te propsy trafiać do tagu `<button>` . 

Kolejna rzecz to, że to jest `const`... Tutaj nie mogłem za ch** zrobić, żeby to był komponent funkcyjny i żeby ustawić mu typ. Jak nie ustawie komponentowi typu `Component` w ten sposób to potem typescript krzyczy przy dodawaniu do komponentu atrybutów dla buttona, które nie są określone w interfejsie.